### PR TITLE
Issue/140 hardware shop item specs

### LINKED ISSUE #140: Add hardware features for hardwareshop
## Description

In the shop, the hardware features of each hardware must be added - see drawing.
**Technical specifications** for:

- Mainboards
- Processors
- Cooler
- RAM
- Graphic cards
- Disks
- Power Packs
- Cases

If you have any questions please contact to me @DevMimas 

Hier die Reihenfolge wie die Hardwaredaten ausgegeben werden. Alle Daten kommen aus [cryptic device / vars.py](https://github.com/cryptic-game/cryptic-device/blob/experimental/app/vars.py)

## CPU
CPU – CoreOne A100 / Fall 1

**Processor - properties**
`Frequency min.: 500 MHz`
`Frequency max.: 500 MHz`
`Socket: XT2019`
`Number cores: 1`
`Max. temperature: 72°C`
`Power usage: 220 W`
**Processor - specific power:**
`Turbospeed: not available`
`Overclock: not available`
**Processor - graphics:**
`Integrated graphics: not integrated`

CPU – DualCore M101 / Fall 2

**Processor - properties**
`Frequency min.: 800 MHz`
`Frequency max.: 800 MHz`
`Socket: XT2020`
`Number cores: 2`
`Max. temperature: 72°C`
`Power usage: 250 W`
**Processor - specific power:**
`Turbospeed: not available`
`Overclock: not available`
**Processor - graphics:**
`Integrated graphics: HD Graphic 100`
`Memory: 512 MB`
`Frequency: 522 MHz`

## Cooler

**Cooler - properties**
`Cooler speed: 1`
`Socket: XT2019`
`Power usage: 10 W`

## RAM

**RAM - properties**
`RAM size: 128 MB`
`Type: DDR 1`
`Frequency: 422 MHz`
`Power usage: 5 W`

## Graphic cards

**Graphic cards - properties**
`RAM size: 128 MB`
`Type: GDDR 1`
`Frequency: 422 MHz`
`Interface: AGP 1`
`Power usage: 80 W`

## Disk

**Disk - properties**
`Typ: HDD`
`Capacity: 2 GB` --> (Achtung! Muss umgerechnet werden!)
`Interface: IDE 1`
`Writing speed: 8 MB/s`
`Reading speed: 16 MB/s`
`Power usage: 15 W`

## Powerpack

**Powerpack - properties**
`Total power: 500 W`

## Case

**Case - properties**
`Name: Mini-ITX`
`Size: small`

## Problem

Without these features, it is impossible to know the characteristics of the hardware such as CPU frequency, RAM size, compatibility, etc.

## Design
Hardwareshop_Technische Daten

### PR DESCRIPTION
**Pull request by SophieErdbeer**
**Description**
Added the technical specifications for the remaining hardware parts for the shop

**Issue**
Closes #140 

**Testing Instructions**
For testing: Open the hardware shop and look at the now added technical specifications

### CODE REVIEW

**Review by JannikEmmerich — CHANGES_REQUESTED** [review]
Please setup your local Git installation correctly. Help can be found [here](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address)

**Review by MarcelCoding — CHANGES_REQUESTED** [review]
Please format your code with the formatter of your IDE.

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Component, Input, OnInit } from '@angular/core';
 import { HardwareShopService } from '../hardware-shop.service';
 import { HardwareShopItem } from '../hardware-shop-item';
 import { animate, state, style, transition, trigger } from '@angular/animations';
-import { Mainboard, PartCategory } from '../../../../api/hardware/hardware-parts';
+import * as hardwareParts from '../../../../api/hardware/hardware-parts';
```
Comment (MarcelCoding) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:5]: Remove `*` import

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 2]
```diff
@@ -127,12 +127,104 @@ export class HardwareShopItemComponent implements OnInit {
...
+           'Memory': cpu.graphicUnit ? cpu.graphicUnit.ramSize : undefined,
+           'Frequency': cpu.graphicUnit ? cpu.graphicUnit.frequency : undefined
+          }
+        };
+
+      case hardwareParts.PartCategory.PROCESSOR_COOLER:
+
+         const processorCooler = this.item.part as hardwareParts.ProcessorCooler;
+
+         return{
```
Comment (MarcelCoding) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:158]: Reformat Code

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 3]
```diff
@@ -127,12 +127,104 @@ export class HardwareShopItemComponent implements OnInit {
...
+            'Frequency' : `${ram.frequency} MHz`,
+            'Power usage' : `${ram.power} W`,
+          }
+        };
+
+      case hardwareParts.PartCategory.GPU:
+
+        const gpu = this.item.part as hardwareParts.GPU;
+
+        return{
```
Comment (MarcelCoding) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:183]: Reformat Code

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 4]
```diff
@@ -127,12 +127,104 @@ export class HardwareShopItemComponent implements OnInit {
...
+            'Interface': gpu.interface,
+            'Power usage': `${gpu.power} W`
+          }
+        };
+
+      case hardwareParts.PartCategory.DISK:
+
+        const disk = this.item.part as hardwareParts.Disk;
+
+        return{
```
Comment (MarcelCoding) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:197]: Reformat Code

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 5]
```diff
@@ -127,12 +127,104 @@ export class HardwareShopItemComponent implements OnInit {
...
+            'Reading speed': `${disk.readingSpeed} MB/s`,
+            'Power usage': `${disk.power} W`
+          }
+        };
+
+      case hardwareParts.PartCategory.POWER_PACK:
+
+        const powerPack = this.item.part as hardwareParts.PowerPack;
+
+        return{
```
Comment (MarcelCoding) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:212]: Reformat Code

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 6]
```diff
@@ -127,12 +127,104 @@ export class HardwareShopItemComponent implements OnInit {
...
+          'Powerpack properties' : {
+            'Total power': `${powerPack.totalPower} W`
+          }
+        };
+
+      case hardwareParts.PartCategory.CASE:
+
+        const _case = this.item.part as hardwareParts.Case;
+
+        return{
```
Comment (MarcelCoding) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:222]: Reformat Code

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

**Review by JannikEmmerich — APPROVED** [review]
Everything that has to do with Git looks fine to me, good job!
The code style can be improved in some places, wait for @MaxiHuHe04 to review for details.

**Review by MarcelCoding — APPROVED** [review]

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 7]
```diff
@@ -127,12 +127,100 @@ export class HardwareShopItemComponent implements OnInit {
...
       case PartCategory.CPU:
-        // const cpu = this.item.part as CPU;
-        // TODO: CPU specifications
-        return {};
+        const cpu = this.item.part as CPU;
+
+        return {
+          'Processor properties': {
+            'Frequeny min': `${cpu.frequencyMin} MHz`,
+            'Frequeny max': `${cpu.frequencyMax} MHz`,
```
Comment (MaxiHuHe04) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:137]: Typos: Frequeny -> Frequency

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 8]
```diff
@@ -127,12 +127,100 @@ export class HardwareShopItemComponent implements OnInit {
...
+            'Cooler speed': processorCooler.coolerSpeed,
+            'Socket': processorCooler.socket,
+            'Power usage': `${processorCooler.power} W`
+          }
+        };
+
+      case PartCategory.RAM:
+        const ram = this.item.part as RAM;
+
+        console.log(ram.ramTyp);
```
Comment (MaxiHuHe04) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:168]: Remaining debug output

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 9]
```diff
@@ -105,7 +105,7 @@ export class HardwareShopItemComponent implements OnInit {
           'Mainboard memory': {
             'Slots': mainboard.ram.ramSlots,
             'Maximum amount': `${mainboard.ram.maxRamSize} MB`,
-            'Type': mainboard.ram.ramTyp.map(type => type.join('')).join(', '),
+            'Type': mainboard.ram.ramTyp.map(type => type.join(' ')).join(', '),
```
Comment (MaxiHuHe04) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:108]: Unlike Mimas said on Discord, DDRx is written without a space 😄

File: src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts [hunk 10]
```diff
@@ -127,12 +127,100 @@ export class HardwareShopItemComponent implements OnInit {
...
+        };
 
-      // TODO: other specifications
+      case PartCategory.GPU:
+        const gpu = this.item.part as GPU;
+
+        return {
+          'Graphic cards properties': {
+            'RAM size': `${gpu.ramSize} MB`,
+            'Type': gpu.ramTyp.join(' '),
```
Comment (MaxiHuHe04) [inline src/app/desktop/windows/hardware-shop/hardware-shop-item/hardware-shop-item.component.ts:185]: Unlike Mimas said on Discord, GDDRx is written without a space 😄

**Review by MaxiHuHe04 — CHANGES_REQUESTED** [review]

**Review by MaxiHuHe04 — APPROVED** [review]
Good job!

**Pull request merged**